app/src/main/res/values/dimen.xml

```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="message_activity_padding">16dp</dimen>
+    <dimen name="spinner_padding">8dp</dimen>
+    <dimen name="message_layout_width">0dp</dimen>
```

Comment:
Do not define zero. It's for layout weight, does not make sense width = 0dp. Let it be 0dp into layout resources file.